reactor3/src/test/java/com/linecorp/armeria/common/reactor3/RequestContextPropagationFluxTest.java

```diff
@@ -0,0 +1,466 @@
+    }
+
+    @Test
+    void connectableFlux() {
+        final ClientRequestContext ctx = newContext();
+        final Flux<String> flux;
+
+        final ConnectableFlux<String> connectableFlux = Flux.just("foo").publish();
+        flux = addCallbacks(connectableFlux.autoConnect(2).publishOn(Schedulers.single()), ctx);
+
+        flux.subscribe().dispose();
+        StepVerifier.create(flux, initialReactorContext(ctx))
+                    .expectSubscriptionMatches(s -> ctxExists(ctx))
+                    .expectNextMatches(s -> ctxExists(ctx) && "foo".equals(s))
+                    .verifyComplete();
+    }
+
+    @Test
+    void connectableFlux_dispose() throws InterruptedException {
+        final ClientRequestContext ctx = newContext();
+        final Flux<String> flux;
+
+        final ConnectableFlux<String> connectableFlux = Flux.just("foo").publish();
+        flux = addCallbacks(connectableFlux.autoConnect(2, disposable -> {
+            assertThat(ctxExists(ctx)).isTrue();
+        }).publishOn(Schedulers.newSingle("aaa")), ctx);
+
+        final Disposable disposable1 = flux.subscribe();
+        await().pollDelay(Duration.ofMillis(200)).until(() -> !disposable1.isDisposed());
+        final Disposable disposable2 = flux.subscribe();
+        await().untilAsserted(() -> {
+            assertThat(disposable1.isDisposed()).isTrue();
+            assertThat(disposable2.isDisposed()).isTrue();
+        });
+    }
+
+    @Test
+    void subscriberContextIsNotMissing() {
+        final ClientRequestContext ctx = newContext();
+        final Flux<String> flux;
+
+        flux = Flux.deferContextual(reactorCtx -> {
+            assertThat((String) reactorCtx.get("foo")).isEqualTo("bar");
+            return Flux.just("baz");
+        });
+
+        final Flux<String> flux1 = flux.contextWrite(reactorCtx -> reactorCtx.put("foo", "bar"));
+        StepVerifier.create(flux1, initialReactorContext(ctx))
+                    .expectSubscriptionMatches(s -> ctxExists(ctx))
+                    .expectNextMatches(s -> ctxExists(ctx) && "baz".equals(s))
+                    .verifyComplete();
+    }
+
+    @Test
+    void ctxShouldBeCleanUpEvenIfErrorOccursDuringReactorOperationOnSchedulerThread() {
+        // Given
+        final ClientRequestContext ctx = newContext();
+        final Flux<String> flux;
+
+        // When
+        flux = Flux.just("Hello", "Hi")
+                   .delayElements(Duration.ofMillis(1000))
+                   .map(s -> {
+                       if (s.equals("Hello")) {
+                           throw new RuntimeException();
+                       }
+                       return s;
+                   })
+                   .contextWrite(Context.of(RequestContextAccessor.accessorKey(), ctx));
+
+        // Then
+        StepVerifier.create(flux)
+                    .expectError(RuntimeException.class)
+                    .verify();
+
+        assertThat(ctxExists(ctx)).isFalse();
```

Comment:
We don't push the RequestContext to the thread local of the current thread that executes this test, which is a Gradle executor, so this never be true.
I'm not sure about the purpose of this test case. 😓 What do you want to achieve using this test?

Reply:
I want to prove that there is no `ThreadLocal` pollution even if `Error` occurs during `Reactor Operation`.
As you said, in previous unit test, it seems not suitable 😢. 

So i modify it below.

```java
// Given
        final ClientRequestContext ctx = newContext();
        final Flux<String> flux;

        // When
        flux = Flux.just("Hello", "Hi")
                   .subscribeOn(Schedulers.single())
                   .delayElements(Duration.ofMillis(1000))
                   .map(s -> {
                       if (s.equals("Hello")) {
                           throw new RuntimeException();
                       }
                       return s;
                   })
                   .contextWrite(Context.of(RequestContextAccessor.accessorKey(), ctx));

        // Then
        StepVerifier.create(flux)
                    .expectError(RuntimeException.class)
                    .verify();

        Flux<String> toVerifyFlux = Flux.just("Dummy")
                                        .subscribeOn(Schedulers.single())
                                        .doOnNext(s -> assertThat(ctxExists(ctx)).isFalse());

        StepVerifier.create(toVerifyFlux)
                    .expectNext("Dummy")
                    .verifyComplete();
```
1. In `Single Thread Scheduler`, Error will occur during `flux Reactor Operation`.
2. and then, `verifyFlux` will be operated on `Single Thread Scheduler` as well. 

In step2, `verifyFlux` will be operated on same thread which `flux` have been operated. If `ThreadLocal` pollution is existed, `assertThat(ctxExists(ctx)).isTrue()`. 

Actually, it should be always `false`, because [this code](https://github.com/reactor/reactor-core/blob/cc4955fe240e0b42f2f78e72c2c95948beb7b8c0/reactor-core/src/main/java/reactor/core/scheduler/Schedulers.java#L996) will be executed.
There is no value in `Reactor Context`, thus there is no value in `ThreadLocal` as well. 

Test code above is not exactly suitable to prove it as well. 
If this code is meaningless, i'm willing to remove it. if you think that this code is useful to show no `ThreadLocal` pollution even if error occurs during `reactive stream` to developer, it will be good to remain.

What do you think?

Reply:
> I want to prove that there is no ThreadLocal pollution even if Error occurs during Reactor Operation
In Single Thread Scheduler, Error will occur during flux Reactor Operation.

I get it. 😉 
If you want to check if the thread local is polluted or not, I think we should check that in the scheduler, not in the test executor. How about:
```java
final Scheduler single = Schedulers.single();
flux = Flux.just("Hello", "Hi")
           .subscribeOn(single)
           ...
...
final CountDownLatch latch = new CountDownLatch(1);
single.schedule(() -> {
    assertThat(ctxExists(ctx)).isFalse();
    latch.countDown();
});
latch.await();
```